capsules/extra/src/screen_on_led.rs

```diff
@@ -0,0 +1,410 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2025.
+
+//! Display virtual LEDs on a screen.
+//!
+//! This creates virtual LEDs by drawing boxes on the screen. The output looks
+//! roughly like this:
+//!
+//! ```text
+//!     ┌─┐┌─┐┌─┐┌─┐
+//! LED └─┘└─┘└─┘└─┘
+//! ```
+//! The boxes get filled in if the LED is on.
+//!
+//! ## Caveat
+//!
+//! The LED API is based on the GPIO API which is synchronous. Writing to a
+//! screen is not. This means LED events can easily get lost if the screen
+//! operation hasn't finished yet.
```

Comment:
One easy fix to work around this is to store a `dirty` bit in this driver. You'd clear it once you start drawing the new LED state, and set it whenever you get an LED state change. Then, when the `write_complete` callback arrives, you simply redraw if `dirty` is set.